CombinedExDemo/App/demo/src/main/java/com/conversation/demo/ChatActivity.kt

```diff
@@ -0,0 +1,451 @@
+
+                notifyConversationError(error)
+
+                if (reason != null && reason == NRError.ConnectionException) {
+                    notifyConnectionError()
+                }
+            }
+
+            NRError.StatementError ->
+
+                if (error.isConversationError()) {
+                    notifyConversationError(error)
+
+                } else {
+                    notifyStatementError(error)
+                }
+
+            else -> {
+                /*all other errors will be handled here. Demo implementation, displays a toast and
+                  writes to the log.
+                 if needed the error.getErrorCode() and sometimes the error.getReason() can provide
+                 the details regarding the error
+                 */
+                Log.e("App-ERROR", error.toString())
+
+                if (reason != null && reason == NRError.ConnectionException) {
+                    notifyConnectionError()
+                } else {
+                    notifyError(error, "general error: ", Color.DKGRAY)
+                }
+            }
+        }
+    }
+
+    private fun notifyConnectionError() {
+        toast("Connection failure.\nPlease check your connection.")
+    }
+
+    private fun notifyConversationError(error: NRError) {
+        notifyError(error, "Conversation is not available: ", Color.parseColor("#6666aa"))
+    }
+
+    private fun notifyStatementError(error: NRError) {
+        notifyError(error, "statement failure - ", Color.RED)
+    }
+
+    @SuppressLint("ResourceType")
+    private fun notifyError(error: NRError, s: String, i: Int) {
+
+        try {
+            supportFragmentManager.fragments.lastOrNull()?.view?.run {
+                snack(
+                    s + error.reason + ": " + error.description,
+                    4000, -1, Gravity.CENTER, intArrayOf(), i
+                )
+            }
+
+        } catch (ignored: Exception) {
+            toast(s + error.reason + ": " + error.description)
+        }
+    }
+
+    override fun onUrlLinkSelected(url: String) {
+        // sample code for handling given link
+        try {
+            val intent = Intent(Intent.ACTION_VIEW).setData(Uri.parse(url))
+            startActivity(intent)
+        } catch (e: Exception) {
+            Log.w(CONVERSATION_FRAGMENT_TAG, "failed to activate link on default app: " + e.message)
+            toast(this, "activating: $url", Toast.LENGTH_SHORT)
+        }
+
+        Log.w(CONVERSATION_FRAGMENT_TAG, "got link activation while activity is no longer available.\n($url)")
+    }
+
+    //-> previous listener method signature @Override onPhoneNumberNavigation(@NonNull String phoneNumber) {
+    override fun onPhoneNumberSelected(phoneNumber: String) {
+        try {
+            val intent = Intent(Intent.ACTION_DIAL)
+            intent.data = Uri.parse("tel:$phoneNumber")
+            startActivity(intent)
+        } catch (e: ActivityNotFoundException) {
+
```

Comment:
"Log.w(CONVERSATION_FRAGMENT_TAG, "failed to open dialer" + e.message)" ?

Reply:
done